browser/brave_rewards/test/creator_detection_browsertest.cc

```diff
@@ -0,0 +1,349 @@
+        </div>
+      </div>
+    </body>
+    </html>
+  )");
+
+  AddBlankTabAndShow(browser());
+  base::test::TestFuture<std::string> future;
+  TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+  NavigateTo("www.twitch.tv", "/testuser");
+
+  EXPECT_EQ(future.Get<0>(), "twitch#author:testuser");
+}
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, TwitterDetection) {
+  SetRewardsEnabled();
+
+  SetResponseHTML(R"(
+    <!doctype html>
+    <html>
+    <head>
+      <script>
+        addEventListener('load', () => {
+          const userEntities = {
+            '987654321': {
+              screen_name: 'testuser',
+              profile_image_url_https: 'https://twitter.com/img'
+            }
+          }
+
+          function getState() {
+            return { entities: { users: { entities: userEntities } } }
+          }
+
+          Object.assign(document.querySelector('#react-root > div'), {
+            __reactProps$123: { children: { props: { store: { getState } } } }
+          })
+        })
+      </script>
+    </head>
+    <body>
+      <div id="react-root">
+        <div></div>
+      </div>
+    </body>
+    </html>
+  )");
+
+  AddBlankTabAndShow(browser());
+  base::test::TestFuture<std::string> future;
+  TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+  NavigateTo("twitter.com", "/testuser");
+
+  EXPECT_EQ(future.Get<0>(), "twitter#channel:987654321");
+}
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, VimeoDetection) {
+  SetRewardsEnabled();
+
+  SetResponseHTML(R"(
+    <!doctype html>
+    <html>
+    <head>
+      <script type="application/ld+json">
+        [
+          {
+            "@type": "Person",
+            "identifier": "987654321",
+            "name": "Test User",
+            "url": "https://vimeo.com/testuser",
+            "image": "https://vimeo.com/user-avatar"
+          }
+        ]
+      </script>
+    </head>
+    <body>
+    </body>
+    </html>
+  )");
+
+  AddBlankTabAndShow(browser());
+  base::test::TestFuture<std::string> future;
+  TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+  NavigateTo("vimeo.com", "/testuser");
+
+  EXPECT_EQ(future.Get<0>(), "vimeo#channel:987654321");
+}
+
+constexpr std::string_view kYouTubeHTML = R"(
+  <!doctype html>
+  <html>
+  <head>
+    <link rel="canonical" href="/channel/987654321" />
+  </head>
+  <body>
+    <ytd-video-owner-renderer>
+      <a href="/@testuser"></a>
+    </ytd-video-owner-renderer>
+    <div id="avatar">
+      <img src="/user-avatar" />
+    </div>
+  </body>
+  </html>
+)";
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, YouTubeDetection) {
+  SetRewardsEnabled();
+  SetResponseHTML(kYouTubeHTML);
+
+  AddBlankTabAndShow(browser());
+  base::test::TestFuture<std::string> future;
+  TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+  NavigateTo("www.youtube.com", "/@testuser");
```

Comment:
Seems like we should do a "same page" navigation as part of this test since that's what YT does?

Reply:
Sound like we need to cover the both cases.